lib/Conversion/TCFToTCP/TCFToTCP.cpp

```diff
@@ -93,6 +97,33 @@ class ConvertBinaryElementwise : public OpRewritePattern<SourceOp> {
 };
 } // namespace
 
+static LogicalResult
+matchAndRewriteUnaryElementwise(Operation *op, PatternRewriter &rewriter) {
+  if (isa<tcf::ExpOp>(op)) {
```

Comment:
I think we would usually use TypeSwitch for this? But it seems in this case it would just be more typing.

Reply:
Yeah, it would be more typing right now. Also, I want this to look ugly and very "open coded" to emphasize the redundancy and need for a solution :imp: .